packages/sdk-middleware-auth/src/existing-token.js

```diff
@@ -0,0 +1,51 @@
+/* @flow */
+import type {
+  Middleware,
+  MiddlewareRequest,
+  MiddlewareResponse,
+  ExistingTokenMiddlewareOptions,
+  Next,
+} from '../../../types/sdk'
+
+export default function createAuthMiddlewareForExistingToken(
+  options: ExistingTokenMiddlewareOptions | string
+): Middleware {
+  return (next: Next) => (
+    request: MiddlewareRequest,
+    response: MiddlewareResponse
+  ) => {
+    const config = options
+      ? {
+          tokenType:
+            typeof options === 'object' && options !== null && options.tokenType
+              ? options.tokenType
+              : 'Bearer',
+          token: typeof options === 'string' ? options : options.token,
+          force: options.force !== undefined ? options.force : true,
+        }
+      : null
+
+    /** The request will not be modified if:
+     *  1. no argument is passed
+     *  2. an object is passed that doesn't contain a `token`
+     *  3. force is false and authorization header exists
+     */
+    if (
+      config === null ||
+      !config.token ||
+      (((request.headers && request.headers.authorization) ||
+        (request.headers && request.headers.Authorization)) &&
+        config.force === false)
+    ) {
+      return next(request, response)
+    }
+    const requestWithAuth = {
+      ...request,
+      headers: {
+        ...request.headers,
+        Authorization: `${config.tokenType} ${config.token}`,
```

Comment:
What other types can be used here? 🤔

I think to be more generic we can simply accept an "authorization header value" instead of the access token, the type etc.

```js
createAuthMiddlewareWithAuthorizationHeader(`Bearer ${access_token}`)

createAuthMiddlewareWithAuthorizationHeader(`Bearer ${access_token}`, { force: false })
```

Reply:
Took my inspiration from [here](https://tools.ietf.org/html/draft-ietf-oauth-v2-15#section-7.1); could basically be anything. 

Only `Bearer` is supported by our API, (Besides `Basic` used for authentication), but I thought it will be a good future-proofing idea to make it configurable

Reply:
Making it configurable is perfectly fine, that's why I suggested to just let the user define the value itself 😉